server.py

```diff
@@ -33,6 +33,16 @@ def get_shopcarts(id):
     return jsonify(message), rc
 
 
+######################################################################
+# DELETE A PRODUCT FROM A SHOPCART
+######################################################################
+@app.route('/shopcarts/<int:uid>/products/<int:pid>', methods=['DELETE'])
+def delete_product(uid, pid):
+    cart = Shopcart.find(uid)
+    if cart:
+        cart.products.pop(pid, None)
```

Comment:
Although this logic is simple, I think this should be abstracted away into the model. This will let us keep the API the same when we move to persistence.